aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java

```diff
@@ -0,0 +1,200 @@
+        }
+
+    }
+
+    private void verifyIndexAndFieldMappings(boolean verifyNewFields) {
+        ValidatableResponse response = asElasticsearchApiUser()
+            .when()
+            .get(CASE_INDEX_ALIAS)
+            .then()
+            .statusCode(200);
+
+        verifyIndexAndAlias(response);
+        verifyMetadataFields(response);
+        verifyCaseDataFields(response, verifyNewFields);
+    }
+
+    private RequestSpecification asElasticsearchApiUser() {
+        return RestAssured.given(new RequestSpecBuilder()
+                                     .setBaseUri(aat.getElasticsearchBaseUri())
+                                     .build());
+    }
+
+    private void verifyIndexAndAlias(ValidatableResponse response) {
+        response
+            .body("", hasKey(CASE_INDEX_NAME))
+            .body(CASE_INDEX_NAME + ".aliases", hasKey(CASE_INDEX_ALIAS));
+    }
+
+    private void verifyMetadataFields(ValidatableResponse response) {
+        response.root(CASE_INDEX_NAME + ".mappings.case.properties");
+
+        verifyFieldsAndType(response, LONG_FIELD_TYPE, "id");
+        verifyFieldsAndType(response, TEXT_FIELD_TYPE, "jurisdiction", "reference", "state");
+        verifyFieldsAndType(response, DATE_FIELD_TYPE, "last_modified", "created_date");
+        verifyFieldsAndType(response, KEYWORD_FIELD_TYPE, "security_classification");
+    }
+
+    private void verifyCaseDataFields(ValidatableResponse response, boolean verifyNewFields) {
+        response.root(CASE_INDEX_NAME + ".mappings.case.properties.data.properties");
+
+        verifyAddressField(response);
+        verifyFieldsAndType(response, TEXT_FIELD_TYPE, "FixedListField", "MultiSelectListField", "PhoneUKField", "TextAreaField", "TextField");
+        verifyFieldsAndType(response, DATE_FIELD_TYPE, "DateField", "DateTimeField");
+        verifyFieldsAndType(response, NUMBER_FIELD_TYPE, "MoneyGBPField", "NumberField");
+        verifyFieldsAndType(response, KEYWORD_FIELD_TYPE, "EmailField", "YesOrNoField");
+
+        if (verifyNewFields) {
+            verifyFieldsAndType(response, "text", "NewTextField");
+        } else {
+            verifyFieldsDoNotOccurInResponse(response, "NewTextField");
+        }
+    }
+
+    private void verifyAddressField(ValidatableResponse response) {
+        response.appendRoot("AddressUKField.properties");
+        verifyFieldsAndType(response, TEXT_FIELD_TYPE, "AddressLine1", "PostCode", "Country");
+        response.detachRoot("AddressUKField.properties");
+    }
+
+    private void verifyFieldsAndType(ValidatableResponse response, String type, String... fields) {
+        Stream.of(fields).forEach(field -> response
+            .body("", hasKey(field))
+            .body(field + ".type", is(type)));
+    }
+
+    private void verifyFieldsDoNotOccurInResponse(ValidatableResponse response, String... fields) {
+        Stream.of(fields).forEach(field -> response.body("", not(hasKey(field))));
+    }
+
+}
```

Comment:
instead of introducing a new spreadsheet specifically for ES, can't we just use the current one used for the other func tests? Saying this because maintaining spreadsheets is a pain

Reply:
Is it ok to delete the index in that case? As I was cleaning up the index etc, I created a specific one for Elasticsearch

Reply:
done now, using the same spreadsheet